### PR TITLE
JSON API docs

### PR DESCRIPTION
**Pull request by xelav**
The docs now in the raw state. But it has all the stuff I wanted to describe. Also I feel that the text in dire need of proofreading

### CODE REVIEW

File: docs/json-api.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+#### Including joint resources
+
+Related resources by default represented as single ID or list of them. If we want to include related recources as nested objects we should use `include` parameter For example:
+
+`/songs?include=releases.artists,sheets`
+
+Note that sorting and filtering can be applied to nested fields too. They can be access through their resource name with dot:
+
+`/songs?include=releases.artists&filter[releases.artists.name.asc]&filter[releases.name.desc]`
```
Comment (kostmetallist) [inline docs/json-api.md:86]: We need query parameter value (`?param=value`)

File: docs/json-api.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+# Technical Documentation for API
+
+## 1. Motivation
+
+In this section we'll discuss what we actually we want from the ~~backend~~ *reargude*'s API
```
Comment (kostmetallist) [inline docs/json-api.md:5]: Reargu**a**rde

File: docs/json-api.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+All basic resources shall implement classic CRUD (**C**reate, **R**ead, **U**pdate, **D**elete) requests with this fairly common set of HTTP methods.
+
+Below the neat table that I snatched directly from restapitutorial.com:
+
+| HTTP Verb | CRUD           | Collection               | Item                        |
+|-----------|----------------|--------------------------|-----------------------------|
+| POST      | Create         | 201 (Created)            | 405 (Method Not Allowed)    |
+| GET       | Read           | 200 (OK)                 | 200 (OK) or 404 (Not Found) |
+| PUT       | Update/Replace | 405 (Method Not Allowed) | 200 (OK) or 404 (Not Found) |
```
Comment (kostmetallist) [inline docs/json-api.md:19]: Write down here 204 No Content

File: docs/json-api.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+Here we'll describe some general functionality that applies for all available resources, unless it's a some special case.
+
+All basic resources shall implement classic CRUD (**C**reate, **R**ead, **U**pdate, **D**elete) requests with this fairly common set of HTTP methods.
+
+Below the neat table that I snatched directly from restapitutorial.com:
+
+| HTTP Verb | CRUD           | Collection               | Item                        |
+|-----------|----------------|--------------------------|-----------------------------|
+| POST      | Create         | 201 (Created)            | 405 (Method Not Allowed)    |
+| GET       | Read           | 200 (OK)                 | 200 (OK) or 404 (Not Found) |
```
Comment (kostmetallist) [inline docs/json-api.md:18]: Describe 204 No Content for empty collection

File: docs/json-api.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+2. We restrict POST query with specific item ID
+3. GET on collections employs basic tecniques such as pagination, sorting and filtering. They will be described in the sections below
+4. For the time being we can ignore PATCH requests or handle them like PUT with only modified part of the object
+
+#### Pagination
+
+  For pagination we picked the simples kind of it: limit/offset pagination. It needs to parameters:
+
+- `offset` - starting row number. Default value is 0.
+- `limit` - sets maximum size of the retrived collection. Default value is 20. Note: we should hardcode some upper limit.
```
Comment (kostmetallist) [inline docs/json-api.md:35]: Set reasonable max limit

File: docs/json-api.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  
+#### Sorting
+
+`sort_by={field_name}.{desc|asc}`
```
Comment (kostmetallist) [inline docs/json-api.md:39]: Explore more unique separator for `asc`/`desc`

File: docs/json-api.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  - *__after* - all dated after value
+  - *__exact* - exact value
+
+#### Date value format
+
+We'll try to stick to [the 5 laws of API dates and times](http://apiux.com/2013/03/20/5-laws-api-dates-and-times/)
+
+In the nutshell:
+
+- Law #1: Use ISO-8601 for your dates (YYYY-MM-DDTHH:mm:ss.ss±hh:mm, e.g. 1937-01-01T12:00:27.87+00:20)
```
Comment (kostmetallist) [inline docs/json-api.md:70]: Milliseconds should take 3 digit positions

File: docs/json-api.md (hunk 6 above)
Comment (xelav) [reply docs/json-api.md]: Now it's `|` _pal.ochka_ symbol

File: docs/json-api.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+#### Including joint resources
+
+Related resources by default represented as single ID or list of them. If we want to include related recources as nested objects we should use `include` parameter For example:
+
+`/songs?include=releases.artists,sheets`
+
+Note that sorting and filtering can be applied to nested fields too. They can be access through their resource name with dot:
+
+`/songs?include=releases.artists&sort_by=releases.artists.name|asc&filter[releases.name]__exact="Fear of the Dark"`
```
Comment (kostmetallist) [inline docs/json-api.md:87]: Sorry, I didn't notice it when we were discussing filter values. According to the https://github.com/lateinit-apps/guitar-app/pull/26/files#diff-ec3c05927629bd864289763f4e2d962eR48, `__<something>` should be provided with a non-quotes value, but here we have one with double quotes. I think it is better to supply `%20`-delimited strings.

**Pull request merged**